service/external_paysession.go

```diff
@@ -44,3 +51,21 @@ func (service *PaymentService) CreateExternalPaymentJourney(req *http.Request, p
 		return nil, Error, err
 	}
 }
+
+func validateClassOfPayment(costs *[]models.CostResourceRest) error {
+
+	for i, cost := range *costs {
+		//Loop through Class Of Payments on a single resource to check they're the same.
+		for j, classOfPayment := range cost.ClassOfPayment {
+			if classOfPayment[j] != classOfPayment[0] {
```

Comment:
Does this work for the following scenario?
```
classOfPayment[0] = 1
classOfPayment[1] = 2
classOfPayment[2] = 2
```

Reply:
Yes it checks if there's a difference between the [0] value and every value afterwards. So in this instance as soon as it evaluates classOfPayment[1] != classOfPayment[0] it'll return an error.